### PR TITLE
#21 test load

### PR DESCRIPTION
**Pull request by SurinaCS**
Testing for load.py. Covers edge cases including data not found in tables. Amendments to load.py include error handling. Test coverage is 96%.

Addressing issue #21

### CODE REVIEW

File: pipeline/load.py [hunk 1]
```diff
@@ -84,8 +95,14 @@ def insert_related_terms(conn: connect, cursor: curs, extracted_dataframe: pd.Da
...
-    cursor.execute("""SELECT keywords.keywords_id FROM keywords WHERE keyword = %s""", (keyword, ))
-    return cursor.fetchone()['keywords_id']
+
+    cursor.execute(
+        """SELECT keywords.keywords_id FROM keywords WHERE keyword = %s""", (keyword, ))
+    result = cursor.fetchone()
+    if result is None:
+        logging.error("Keyword '%s' not found in the database.", keyword)
+        raise ValueError(f"Keyword '{keyword}' not found.")
+    return result['keywords_id']
```
Comment (keoghrmj) [inline pipeline/load.py:105]: Nice catch

**Review by keoghrmj — APPROVED** [review]
Good tests, type hints and error handling again!! Keep up the great work

File: pipeline/test_load.py [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+    assert mock_curs.execute.call_count == 1
+    assert mock_conn.commit.call_count == 1
+
+
+@patch('load.insert_related_term_assignment')
+@patch('load.insert_related_terms')
+@patch('load.insert_keyword_recordings')
+@patch('load.insert_keywords')
+@patch('load.setup_connection')
+def test_main_success(mock_setup, mock_insert_keywords, mock_insert_recordings, mock_insert_related, mock_insert_assignment, mock_df, env, caplog):
```
Comment (keoghrmj) [inline pipeline/test_load.py:223]: Lovely test
